geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapred/GeoMesaInputFormat.scala

```diff
@@ -0,0 +1,281 @@
+import org.apache.accumulo.core.client.security.tokens.PasswordToken
+import org.apache.accumulo.core.data.{Key, Value}
+import org.apache.accumulo.core.security.Authorizations
+import org.apache.accumulo.core.util.{Pair => AccPair}
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.io.{Text, Writable}
+import org.apache.hadoop.mapred._
+import org.geotools.data.{DataStoreFinder, Query}
+import org.geotools.filter.identity.FeatureIdImpl
+import org.geotools.filter.text.ecql.ECQL
+import org.locationtech.geomesa.core.data.{AccumuloDataStore, AccumuloDataStoreFactory}
+import org.locationtech.geomesa.core.index._
+import org.locationtech.geomesa.feature.FeatureEncoding.FeatureEncoding
+import org.locationtech.geomesa.feature.{ScalaSimpleFeature, SimpleFeatureDecoder}
+import org.locationtech.geomesa.jobs.GeoMesaConfigurator
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+import org.opengis.filter.Filter
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+import scala.collection.mutable.ArrayBuffer
+
+object GeoMesaInputFormat extends Logging {
+
+  /**
+   * Configure the input format.
+   *
+   * This is a single method, as we have to calculate several things to pass to the underlying
+   * AccumuloInputFormat, and there is not a good hook to indicate when the config is finished.
+   */
+  def configure(job: JobConf,
+                dsParams: Map[String, String],
+                featureTypeName: String,
+                filter: Option[String]): Unit = {
+
+    val ds = DataStoreFinder.getDataStore(dsParams).asInstanceOf[AccumuloDataStore]
+
+    assert(ds != null, "Invalid data store parameters")
+
+    // set up the underlying accumulo input format
+    val user = AccumuloDataStoreFactory.params.userParam.lookUp(dsParams).asInstanceOf[String]
+    val password = AccumuloDataStoreFactory.params.passwordParam.lookUp(dsParams).asInstanceOf[String]
+    InputFormatBase.setConnectorInfo(job, user, new PasswordToken(password.getBytes()))
+
+    val instance = AccumuloDataStoreFactory.params.instanceIdParam.lookUp(dsParams).asInstanceOf[String]
+    val zookeepers = AccumuloDataStoreFactory.params.zookeepersParam.lookUp(dsParams).asInstanceOf[String]
+    InputFormatBase.setZooKeeperInstance(job, instance, zookeepers)
+
+    val auths = Option(AccumuloDataStoreFactory.params.authsParam.lookUp(dsParams).asInstanceOf[String])
+    auths.foreach(a => InputFormatBase.setScanAuthorizations(job, new Authorizations(a.split(","): _*)))
+
+    // run an explain query to set up the iterators, ranges, etc
+    val ecql = filter.map(ECQL.toFilter).getOrElse(Filter.INCLUDE)
+    val query = new Query(featureTypeName, ecql)
+    val queryPlans = ds.explainQuery(featureTypeName, query, ExplainNull)
+
+    // see if the plan is something we can execute from a single table
+    val tryPlan = if (queryPlans.length > 1) None else queryPlans.headOption.filter {
```

Comment:
I don't really like the idea of the query planner having any notion of m/r. But we do have to duplicate some code down below to avoid that. Hmm, let me think about it.